### PR TITLE
Wait for node to be Deployed

### LINKED ISSUE #1: Wait for Deployment via API
### Problem

The script currently relies on a `sleep` to wait for the node to finish deploying in Tutum.

### Solution

The `tutum-cli` gives the state of the deployment via:

```
tutum node inspect $TUTUM_UUID
```

This can be used to get a more accurate sleep duration.

### PR DESCRIPTION
**Pull request by jskeates**
### Problem

#1. 

### Solution

Uses coreutils' `timeout`, which comes on Ubuntu by default.

The command times out after 5 minutes, after which the startup script stops (so no tags are added to Tutum). You could also trigger a shutdown here and the node would be automatically replaced by the Autoscaling Group, but would mean you can't then get onto the node to do any debugging.

### CODE REVIEW

**Comment by revett** [comment]
Thanks for the PR @jskeates! Did you see @amegianeg's [comment](https://github.com/vidsy/tutum-startup/issues/1#issuecomment-171268914) however about using `--sync`?

It would be valuable including that in this PR as well I think.

**Comment by jskeates** [comment]
@revett I did, but couldn't work out what to command to append `--sync` to. Maybe @amegianeg has an idea? Am happy to include it if we can work it out.

File: script.sh [hunk 1]
```diff
@@ -36,14 +36,16 @@ tutum node byo | sed -n 4p | source /dev/stdin
...
-sleep 15
-
 # Set Tutum UUID env var now that tutum-agent has been installed
 
 export TUTUM_UUID=$(cat /etc/tutum/agent/tutum-agent.conf | jq -r .TutumUUID)
 
+# Wait for node to be deployed
+
+echo "Waiting for node to be deployed..."
+timeout 5m bash -c "while [ \"\$(tutum node inspect $TUTUM_UUID | jq -r .state)\" != \"Deployed\" ]; do sleep 10; done;" #TUTUM_UUID is purposefully not escaped
```
Comment (revett) [inline script.sh]: Can we make `5m` a variable at the top of the file to make it easier to configure?

**Comment by revett** [comment]
Looks good. Worth waiting until @amegianeg describes how `--sync` works. Also added some comments.

**Comment by jskeates** [comment]
Once #7 is merged, I'll add that variable for the timeout (don't want to add a Parameters section twice in two PRs)

**Comment by amegianeg** [comment]
Sorry for the delay guys :(
So reviewing CLI code I made a mistake: `tutum node byo` is not implemented the way I thought (registering the own host itself in Tutum's API) so the `--sync` flag does not work for this command (it just builds the command and prompts it) so unfortunately .

**Comment by revett** [comment]
#7 has now been merged.

**Comment by jskeates** [comment]
Updated with the variable

**Comment by revett** [comment]
:cake:

**Pull request closed without merging**